src/v/cloud_storage/partition_manifest.cc

```diff
@@ -399,7 +452,29 @@ partition_manifest::difference(const partition_manifest& remote_set) const {
       end(),
       remote_set.begin(),
       remote_set.end(),
-      result.get_insert_iterator());
+      conditional_inserter(
+        [&remote_set](const auto& kv) {
+            // Insert 'replaced' only if it's not present in
+            // the remote_set manifest (represents new data)
+            auto it = remote_set._replaced.find(kv.first);
```

Comment:
Isn't this `set_difference` meant to calculating the difference between the main list of segments, and the next one acting on the `_replaced` segments?  I'm confused why the condition for this one is looking at `_replaced_ but updating `_segments`.